cmd/ocm/create/machinepool/cmd.go

```diff
@@ -0,0 +1,165 @@
+  ocm create machinepool --cluster=mycluster --labels="foo=bar,bar=baz" mp-1`,
+	RunE: run,
+}
+
+func init() {
+	flags := Cmd.Flags()
+
+	flags.StringVarP(
+		&args.clusterKey,
+		"cluster",
+		"c",
+		"",
+		"Name or ID of the cluster to add the machine pool to (required).",
+	)
+	//nolint:gosec
+	Cmd.MarkFlagRequired("cluster")
+
+	flags.StringVar(
+		&args.instanceType,
+		"instance-type",
+		"",
+		"Instance type that should be used.",
+	)
+
+	flags.IntVar(
+		&args.replicas,
+		"replicas",
+		0,
+		"Count of machines for this machine pool.",
+	)
+
+	//nolint:gosec
+	Cmd.MarkFlagRequired("replicas")
+
+	flags.StringVar(
+		&args.labels,
+		"labels",
+		"",
+		"Labels for machine pool. Format should be a comma-separated list of 'key=value'. "+
+			"This list will overwrite any modifications made to Node labels on an ongoing basis.",
+	)
+}
+
+func run(cmd *cobra.Command, argv []string) error {
+
+	// Check that the cluster key (name, identifier or external identifier) given by the user
+	// is reasonably safe so that there is no risk of SQL injection:
+	clusterKey := args.clusterKey
+	if !c.IsValidClusterKey(clusterKey) {
+		return fmt.Errorf(
+			"Cluster name, identifier or external identifier '%s' isn't valid: it "+
+				"must contain only letters, digits, dashes and underscores",
+			clusterKey,
+		)
+	}
+
+	if len(argv) < 1 || argv[0] == "" {
+		return fmt.Errorf("Missing machine pool ID")
+	}
+	machinePoolID := argv[0]
+
+	labels := make(map[string]string)
+	if args.labels != "" {
+		for _, label := range strings.Split(args.labels, ",") {
+			if !strings.Contains(label, "=") {
+				return fmt.Errorf("Expected key=value format for label-match")
+			}
+			tokens := strings.Split(label, "=")
+			labels[strings.TrimSpace(tokens[0])] = strings.TrimSpace(tokens[1])
+		}
+	}
+
+	// Create the client for the OCM API:
+	connection, err := ocm.NewConnection().Build()
+	if err != nil {
+		return fmt.Errorf("Failed to create OCM connection: %v", err)
+	}
+	defer connection.Close()
+
+	// Get the client for the cluster management api
+	clusterCollection := connection.ClustersMgmt().V1().Clusters()
+
+	cluster, err := c.GetCluster(clusterCollection, clusterKey)
+	if err != nil {
+		return fmt.Errorf("Failed to get cluster '%s': %v", clusterKey, err)
+	}
+
+	if cluster.State() != cmv1.ClusterStateReady {
+		return fmt.Errorf("Cluster '%s' is not yet ready", clusterKey)
+	}
+
+	_, err = c.ValidateMachineType(
+		connection.ClustersMgmt().V1(),
+		cluster.CloudProvider().ID(),
+		args.instanceType)
+	if err != nil {
+		return fmt.Errorf("Expected a valid machine type: %s", err)
+	}
+
+	machinePoolBuilder := cmv1.NewMachinePool()
+
+	machinePoolBuilder = machinePoolBuilder.ID(machinePoolID)
+	machinePoolBuilder = machinePoolBuilder.InstanceType(args.instanceType)
+	machinePoolBuilder = machinePoolBuilder.Replicas(args.replicas)
+
+	if len(labels) > 0 {
+		machinePoolBuilder = machinePoolBuilder.Labels(labels)
+	}
```

Comment:
Nitpick: This build process can be refactored to:
```
machinePoolBuilder.ID(machinePoolID).
    InstanceType(args.instanceType).
    Replicas(args.replicas).
    Labels(labels)
```

As far as I remember its fine for the labels to be an empty.